pkg/expressions/json_templates_test.go

```diff
@@ -284,6 +284,166 @@ value13: | # This is a string
 				require.Equal(t, "42\n", parsed["value13"])
 			},
 		},
+		{
+			name:         "standard delimiter with missing closing tag",
+			jsonTemplate: `{ "AString": "${{ aString " }`,
+			assertions: func(t *testing.T, _ []byte, err error) {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), "Cannot find end tag")
+			},
+		},
+		{
+			name:         "standard delimiter cannot handle }} in expression body",
+			jsonTemplate: `{ "AString": "${{ quote(\"{{.domain}}\") }}" }`,
```

Comment:
I would try to leave `quote()` out of the tests if I were you. Despite how the issue was originally reported, you'll see I had quickly concluded that using the `quote()` function wasn't a factor. Knowing that, if we don't involve `quote()`, these tests are all much easier to understand.